subsys/fs/fs.c

```diff
@@ -168,6 +168,25 @@ int fs_open(struct fs_file_t *zfp, const char *file_name, fs_mode_t flags)
 		return rc;
 	}
 
+	if (flags & FS_O_TRUNC) {
+		if ((flags & FS_O_WRITE) == 0) {
+			/** Truncate not allowed when file is not opened for write */
+			LOG_ERR("file should be opened for write to truncate!!");
+			return -EACCES;
+		}
+		CHECKIF(mp->fs->truncate == NULL) {
+			LOG_ERR("file truncation not supported!!");
+			return -ENOTSUP;
+		}
```

Comment:
This part should be directly tested with the check for `FS_O_TRUNC`, fail if you can predict failure before doing anything.

Reply:
A good catch!